### DISCUSSION TITLE
Iperf3 UDP bi-dir packet counts

### DISCUSSION THREAD

**Discussion by djamp42** [esnet/iperf#1268] [Q&A]
I looked around but couldn't find anything explaining this.. after a bi-dir udp test i see the following..
sum_sent_bidir_reverse
                        "lost_packets": 0,
                        "packets":      108676,

"sum_received_bidir_reverse": 
                        "lost_packets": 78937,
                        "packets":      107183,

I did loose packets on this, but don't know how it's getting that lost_packets number **78937**

108676 - 107183 = **1943** is what i would expect to see.

I just built this from the latest github, because i needed the json output for bidir test :)

**Comment by davidBar-On** [comment]
When using bi-dir, the sending streams on both sides are independent.  It is **not** that the server sends back the packets received from the client.  Therefore, number of packets on both streams are independent (although they are expected to be similar, as they usually traverse similar path but on opposite directions).

**Reply by bmah888** [reply]
@davidBar-On is correct that the streams are independent. Data goes in both directions simultaneously.

You could have somewhat different numbers on the two directions due to asymmetric routing (which is actually pretty common in wide-area scenarios) or asymmetric links (which are common in many residential broadband deployments).
